cmd/juju/storage/remove.go

```diff
@@ -0,0 +1,140 @@
+// Copyright 2016 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package storage
+
+import (
+	"strings"
+
+	"github.com/juju/cmd"
+	"github.com/juju/errors"
+	"gopkg.in/juju/names.v2"
+
+	"github.com/juju/juju/apiserver/params"
+	"github.com/juju/juju/cmd/juju/common"
+	"github.com/juju/juju/cmd/modelcmd"
+)
+
+// NewRemoveStorageCommandWithAPI returns a command
+// used to remove storage from the model.
+func NewRemoveStorageCommandWithAPI() cmd.Command {
```

Comment:
Agreed that the current implementation is messy. Was aiming for consistency initially, as the current code base if full of 10 ways to do the same thing. A tech debt PR could make the bulk changes required later. But didn't block on this.